### ISSUE TITLE
@stylistic/block-opening-brace-newline-before fix produces unparsable code after an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#89]
Found while fixing #63, by sweeping every rule that deletes a trailing whitespace run from a `raws` string. Unlike #63 and #87 the comment itself survives here — what is lost is the newline that terminates it, and with it the whole stylesheet.

### What minimal example or steps are needed to reproduce the bug?

```scss
.a // keep me
{
	color: red;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/block-opening-brace-newline-before": "never-multi-line"
  }
}
```

### What did you expect to happen?

An inline comment ending the selector makes the rule impossible to satisfy: whatever is put on that line after the comment is part of the comment. The fix should therefore leave the code alone and let the warning stand, rather than rewrite the source into something that no longer parses.

### What actually happened?

```scss
.a // keep me{
	color: red;
}
```

The opening brace is now inside the comment, so the block is no longer a block. Parsing the output back fails with `<css input>:2:2: Unknown word color`, and every subsequent tool in the pipeline sees a broken stylesheet.

The cause is at [`lib/rules/block-opening-brace-newline-before/index.js:93`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/block-opening-brace-newline-before/index.js#L93):

```js
else if (primary.startsWith(`never`)) statement.raws.between = statement.raws.between.replace(/\s*$/u, ``)
```

`\s*$` treats the newline that closes the `//` comment as ordinary trailing whitespace. Block comments are handled correctly, since nothing about them depends on a newline — only the inline form is affected. The `always` branch is safe too: it inserts a newline and deletes nothing.

### Do you have a proposal to fix the bug?

The same shape as the fix for #63: cut the trailing whitespace run off `between`, and when what remains ends with an inline comment, put a newline back — here that means leaving `between` as it was and letting the warning stand, since `never` cannot be satisfied at all in that position.

Once a second rule needs this reasoning — #87 and #88 do — the check for “the content ends with an inline comment” is worth extracting into a shared helper in `lib/utils/` instead of being spelled out in every rule.
